File: leetcode/java/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target){
        HashMap<Integer, Integer> seen = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int currNum = nums[i];
            int secondNum = target - currNum;
            if(seen.containsKey(secondNum)){
                return new int[]{seen.get(secondNum), i};
            }
            seen.put(currNum, i);
        }
        return new int[]{};
    }
}

File: leetcode/javascript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const seen = {};
    for(let i = 0; i < nums.length; i++){
        const firstNum = nums[i];
        const secondNum = target - firstNum;
        if(secondNum in seen){
            return [seen[secondNum], i];
        }
        seen[firstNum] = i;
    }
    return [0, 0];
};